### ISSUE TITLE
Can we use some static analysis like Sonar in CI on PR?

### ISSUE THREAD

**Issue by c71n93** [objectionary/eo#3242]
I faced several problems while fixing issues with [Sonar Cloud](https://sonarcloud.io/). The main problem is that Sonar Cloud doesn't fit in our organization's workflow. We can't use it to check new code from PRs, because of the problem with repository secrets and forks. According to https://docs.github.com/en/actions/security-guides/using-secrets-in-github-actions: "With the exception of `GITHUB_TOKEN`, secrets are not passed to the runner when a workflow is triggered from a forked repository."

So now, we have situation, where Sonar analysis is running in new code only after it is merged into master. It means, that we will find out about the bugs when they are already in the master. So this problem is obvious. 

Another problem appears when someone is trying to fix such bugs. This person need to make sure, that problem is actually fixed, but the only straightforward way to do it is merge code fixes this problems (or not). In my case I created new account on Sonar Cloud and ran analysis on my forked repo with the changes I need to test (example: https://github.com/c71n93/eo/pull/5). It's an ugly way, but I haven't found another one.

Thus, currently all new Sonar violations are simply ignored. Getting rid of Sonar is definitely not the way. I think the only way to overcome this problems is to add some other static analysis checks on PR, that will cover most of Sonar checks (because the Sonar analysis violations are really useful).

**Comment by yegor256** [comment]
@c71n93 good description of the problem, thanks!

**Comment by yegor256** [comment]
@c71n93 what about this plugin: https://github.com/marketplace/actions/sonarqube-quality-gate-check maybe we can use it? Also, here is some relevant info: https://docs.sonarsource.com/sonarqube/latest/devops-platform-integration/github-integration/adding-analysis-to-github-actions-workflow/

**Comment by 0crat** [comment]
@c71n93 Great job reporting that new bug! 🐛 As per our work policy, you've earned **8** points for your keen eye. Your contribution directly improves our project's quality. Remember, if you spot anything else that [doesn't look](https://www.yegor256.com/2018/02/06/where-to-find-more-bugs.html) quite right, [don't hesitate](https://www.yegor256.com/2014/04/13/bugs-are-welcome.html) to [report](https://www.yegor256.com/2018/04/24/right-way-to-report-bugs.html) it. Each bug report is valuable and rewarded!

**Comment by volodya-lombrozo** [comment]
@yegor256 I suggest closing this one, because it's a security limitation of GA. Thus, yes, the only way we can check our repo using Sonar is just after a PR is merged and then fix it.
